syntax = "proto3";
package rpc_server;

service LoginServer
{
    rpc Login (LoginReq) returns (LoginRes);   // 登录服务
    rpc Logout (LogoutReq) returns (LogoutRes);  // 登出服务
    rpc Register (RegisterReq) returns (RegisterRes);  // 注册服务
    rpc Authenticate (AuthenticateReq) returns (AuthenticateRes);  // 令牌验证服务
    rpc Change_password (ChangePasswordReq) returns (ChangePasswordRes);  // 修改密码服务
    rpc Get_online_users (GetOnlineUsersReq) returns (GetOnlineUsersRes); // 获取在线用户列表
}

// 登录服务请求
message LoginReq
{
    bytes account = 1;  // 账号
    bytes password = 2; // 密码：RSA加密
}
// 登录服务响应
message LoginRes
{
    bool success = 1;   // 操作结果
    bytes message = 2; // 响应信息
    bytes token = 3;  // 会话或令牌
}

// 登出服务请求
message LogoutReq
{
    bytes token = 1;
}
// 登出服务响应
message LogoutRes
{
    bool success = 1;
    bytes message = 2;
}

// 注册服务请求
message RegisterReq
{
    bytes user_name = 1;
    bytes password = 2;
    bytes email = 3;
}
// 注册服务响应
message RegisterRes
{
    bool success = 1;
    bytes message = 2;
}

// 令牌验证服务请求
message AuthenticateReq
{
    bytes account = 1;
    bytes token = 2;
}
// 令牌验证服务响应
message AuthenticateRes
{
    bool success = 1;
    bytes message = 2;
}

// 修改密码服务请求
message ChangePasswordReq
{
    bytes token = 1;
    bytes old_password = 2;     // 原密码：RSA加密
    bytes new_password = 3;     // 新密码：RSA加密
}
// 修改密码服务响应
message ChangePasswordRes
{
    bool success = 1;
    bytes message = 2;
}

// 获取在线用户列表请求
message GetOnlineUsersReq
{
    bytes token = 1;
}
// 获取在线用户列表响应
message GetOnlineUsersRes
{
    bool success = 1;
    bytes message = 2;
    repeated bytes users = 3;
}